rust/examples/otel-stdout-components.rs

```diff
@@ -0,0 +1,92 @@
+use dylibso_observe_sdk::{adapter::otelstdout::OtelStdoutAdapter, context::component::ObserveSdkBindings};
+
+use wasmtime::component::Val;
+use wasmtime_wasi::preview2::{WasiView, WasiCtx, Table};
+
+struct State {
+    table: Table,
+    wasi_ctx: WasiCtx,
+    observe_sdk_bindings: ObserveSdkBindings,
+}
+
+impl WasiView for State {
+    fn table(&self) -> &Table {
+        &self.table
+    }
+
+    fn table_mut(&mut self) -> &mut Table {
+        &mut self.table
+    }
+
+    fn ctx(&self) -> &WasiCtx {
+        &self.wasi_ctx
+    }
+
+    fn ctx_mut(&mut self) -> &mut WasiCtx {
+        &mut self.wasi_ctx
+    }
+}
+
+impl AsMut<ObserveSdkBindings> for State {
+    fn as_mut(&mut self) -> &mut ObserveSdkBindings {
+        &mut self.observe_sdk_bindings
+    }
+}
+
+#[tokio::main]
+pub async fn main() -> anyhow::Result<()> {
+    let args: Vec<_> = std::env::args().skip(1).collect();
+    let data = std::fs::read(&args[0])?;
+    let function_name = "hello-world";
+    let mut config = wasmtime::Config::new();
+
+    config.async_support(true);
```

Comment:
I had to turn on async support to get around an assertion in wasmtime – I'm _pretty sure_ it's because Wasi preview2 requires async, but not certain.

In any case, enable it here.